### ISSUE TITLE
Fingerprint definition

### ISSUE THREAD

**Issue by hkmztrk** [cdk/cdk-build-util#4]
Hello,

I'd like to ask that whether there is a paper reference for Extended Fingerprint that explains how the fingerprints are decided.
(https://cdk.github.io/cdk/2.0/docs/api/org/openscience/cdk/fingerprint/ExtendedFingerprinter.html)

Thanks!

**Comment by johnmay** [comment]
Not exactly, but it's your bog standard path based (aka Daylight fingerprint): http://www.daylight.com/dayhtml/doc/theory/theory.finger.html. It does set some extra bits for ring sizes, but I don't know if there were tested when it was originally or how much these help etc. For example you can actually capture ring while you traverse the paths, obviously if you can reach back to the start point you know it's a ring and you can hash it differently.

BTW You should only be using it for substructure searching :-).

**Comment by johnmay** [comment]
Will leave it open unless there is actually something to extra needed...

**Comment by hkmztrk** [comment]
Thank you.

> BTW You should only be using it for substructure searching :-).

I was using it for ligand representation and to compute similarities.

**Comment by johnmay** [comment]
Use CircularFingerprint (ECFP4) it performs much better for similarity, AKA morgan fingerprint in RDKIT. Old poster but shows performance on well known benchmark Briem Lessel: https://chemaxon.com/app/uploads/2011/05/NextMovePoster3.pdf
